### PR TITLE
Add OAuth login support

### LINKED ISSUE #2: Discussion: OAuth instead of PAT
## Description

The login process could be simplified by implementing OAuth instead of requiring the user to generate a PAT.

Looking at [this line](https://git.io/JTaa0) in the GitHub CLI, it seems perfectly safe to embed OAuth credentials in a CLI.

### LINKED ISSUE #22: Ignores Absolute path for structure_root
## Description

Ignores absolute path in `structure_root`

## Steps to reproduce

Set:

`structure_root: "/mnt/repos"`

in your `~/.config/fgh/config.yaml`

Then:

`gfh clone  Matt-Gleich/fgh`

The project is checked out at `/home/username/mnt/repos`

I was expecting it to clone at:

`/mnt/data/`

### PR DESCRIPTION
**Pull request by cjdenio**
## Description

Three cheers for OAuth! :tada: 

## Steps

- [x] My change requires a change to the documentation
- [ ] I have updated the accessible documentation according
- [x] I have read the **CONTRIBUTING.md** file
- [x] There is no duplicate open or closed pull request for this fix/addition/issue resolution.

## Original Issue

This PR resolves #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,12 +54,13 @@ As you begin contributing to an increasing amount of GitHub repositories, you'll
 
 ## 📚 Usage
 
-### ⚙️ `fgh configure`
+### 🔒 `fgh login`
+
+Before using `fgh`, you'll need to give it access to your GitHub account. Simply run `fgh login` to quickly get set up!
```
Comment (Matt-Gleich) [inline README.md:59]: Could we add a note saying that users could use a PAT if they can't get OAuth access (organizations permissions for example)? Is it a good idea to point out where the token is stored on a user's machine?

File: pkg/commands/login/exchange.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package login
+
+import (
+	"context"
+	"time"
+)
+
+func Exchange(code string) (string, error) {
```
Comment (Matt-Gleich) [inline pkg/commands/login/exchange.go:8]: Please add godoc to all functions you created

**Review by Matt-Gleich — CHANGES_REQUESTED** [review]
2 small suggestions but besides that, I think we are good to merge! Really great job on this. The code is super clear, organized, and well documented. Thank you so much!!

**Review by Matt-Gleich — APPROVED** [review]
Looks good to me!

**Comment by Matt-Gleich** [comment]
Thanks again @cjdenio!!

**Comment by cjdenio** [comment]
No problem! Glad to help!

**Pull request merged**